### ISSUE TITLE
Add magnitude-based gate initialization

### ISSUE THREAD

**Issue by MaxGhenis** [PolicyEngine/L0#8]
## Description
Initialize L0 gates based on weight magnitudes, so that larger weights are more likely to be kept initially.

## Motivation
Weight magnitude is often correlated with importance. Starting with magnitude-informed gates could lead to faster convergence and better final models.

## Proposed Implementation
```python
def magnitude_init(weights, target_sparsity=0.5):
    """Initialize gates based on weight magnitudes."""
    magnitudes = weights.abs()
    threshold = torch.quantile(magnitudes, target_sparsity)
    
    # Initialize gates: high probability for large weights
    init_probs = torch.sigmoid((magnitudes - threshold) / threshold * 5)
    logits = torch.log(init_probs / (1 - init_probs + 1e-8))
    return logits

# Usage
layer = L0Linear(100, 50, init_method='magnitude')
```

## Options
- Magnitude-based
- Gradient-based (after initial training)
- Fisher information-based
